src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs

```diff
@@ -0,0 +1,324 @@
+			var item = dragEventArgs.DataView.FindRawData(ReorderItemFormatId);
+			var container = dragEventArgs.DataView.FindRawData(ReorderContainerFormatId) as FrameworkElement; // TODO: This might have changed/been recycled if scrolled 
+			if (that is null || src is null || item is null || container is null || src != that)
+			{
+				dragEventArgs.Log().Warn("Invalid reorder event.");
+
+				return;
+			}
+
+			that.UpdateReordering(dragEventArgs.GetPosition(that), container, item);
+		}
+
+		private static void OnReorderCompleted(object sender, _DragEventArgs dragEventArgs)
+		{
+			var that = sender as ListView;
+			var src = dragEventArgs.DataView.FindRawData(ReorderOwnerFormatId) as ListView;
+			var item = dragEventArgs.DataView.FindRawData(ReorderItemFormatId);
+			var container = dragEventArgs.DataView.FindRawData(ReorderContainerFormatId) as FrameworkElement; // TODO: This might have changed/been recycled if scrolled 
+			if (that is null || src is null || item is null || container is null || src != that)
+			{
+				dragEventArgs.Log().Warn("Invalid reorder event.");
+
+				return;
+			}
+
+			var updatedIndex = default(Uno.UI.IndexPath?);
+			that.CompleteReordering(container, item, ref updatedIndex);
+
+			if (that.IsGrouping
+				|| !updatedIndex.HasValue
+				|| !(dragEventArgs.DataView.FindRawData(DragItemsFormatId) is List<object> movedItems))
+			{
+				return;
+			}
+
+			var unwrappedSource = that.UnwrapItemsSource();
+			switch (unwrappedSource)
+			{
+				case null: // The ListView was created with items defined in XAML
+					var items = that.Items;
+					ProcessMove(
+						items.Count,
+						items.IndexOf,
+						(oldIndex, newIndex) =>
+						{
+							var item = items[oldIndex];
+							items.RemoveAt(oldIndex);
+							if (newIndex >= items.Count)
+							{
+								items.Add(item);
+							}
+							else
+							{
+								items.Insert(newIndex, item);
+							}
+						});
+					break;
+
+				case IObservableVector vector:
+					ProcessMove(
+						vector.Count,
+						vector.IndexOf,
+						(oldIndex, newIndex) =>
+						{
+							var item = vector[oldIndex];
+							vector.RemoveAt(oldIndex);
+							if (newIndex >= vector.Count)
+							{
+								vector.Add(item);
+							}
+							else
+							{
+								vector.Insert(newIndex, item);
+							}
+						});
+					break;
+
+				default:
+					// The UWP ListView seems to automatically push back changes only if the ItemsSource inherits from ObservableCollection
+					var srcType = unwrappedSource.GetType();
+					if (srcType.IsGenericType
+						&& srcType.GetGenericTypeDefinition() == typeof(ObservableCollection<>)
+						&& srcType.GetMethod(nameof(ObservableCollection<object>.Move), new[] { typeof(int), typeof(int) }) is { } mv)
+					{
+						ProcessMove(
+							((ICollection)unwrappedSource).Count,
+							((ICollection)unwrappedSource).IndexOf,
+							(oldIndex, newIndex) => mv.Invoke(unwrappedSource, new object[] { oldIndex, newIndex }));
```

Comment:
😬

Reply:
Yup ... but there is no untyped version :/ As this is not expected to be invoked too much, it's acceptable. I've forgotten however to ensure that the linker does not remove it 😱 Gonna add it.

Reply:
Finally I just removed the reflection: UWP does remove + insert even on `ObservableCollection`